kernel/kernel-api/src/main/java/io/delta/kernel/internal/SnapshotImpl.java

```diff
@@ -60,11 +60,32 @@ public SnapshotImpl(
     this.inCommitTimestampOpt = Optional.empty();
   }
 
+  /////////////////
+  // Public APIs //
+  /////////////////
+
   @Override
   public long getVersion(Engine engine) {
     return version;
   }
 
+  @Override
+  public long getTimestamp(Engine engine) {
+    if (TableConfig.isICTEnabled(engine, metadata)) {
```

Comment:
I'm wondering if maybe this should be part of the snapshot creation (when we also load protocol & metadata)?

Reply:
`getTimestamp` also kind of implies that we don't do any real work to retrieve this

Reply:
> I'm wondering if maybe this should be part of the snapshot creation (when we also load protocol & metadata)?

That's a great point! Let me re-think this ...

Reply:
This will be a decent amount of work to do this. I wish this had been done with ICT was implemented in Kernel.

@vkorukanti any thoughts here?

Reply:
I just spent several hours implementing this, and realized that CommitInfo isn't stored in the checkpoint ... so, loading ICT from logReplay would only work when the latest log segment is a delta file not a checkpoint. darn.

Reply:
Maybe as a future optimization we can get the `inCommitTimestampOpt` as part of log replay if the latest log segment is a delta file, and if not leave it empty. But i'm fine if this PR does not make any changes for that.